packages/salesforce-adapter/test/change_validators/flow_referenced_elements.test.ts

```diff
@@ -0,0 +1,163 @@
+              connector: { targetReference: 'Decision' },
+            },
+          ],
+          decisions: [
+            {
+              name: 'Decision',
+              connector: { targetReference: 'RecordCreate' },
+            },
+          ],
+          recordCreates: [
+            {
+              name: 'RecordCreate',
+            },
+          ],
+        },
+        Flow,
+      )
+      flowChange = toChange({ after: flow })
+    })
+    it('should not return any errors', async () => {
+      const errors = await flowReferencedElements([flowChange])
+      expect(errors).toHaveLength(0)
+    })
+  })
+  describe('when there are references to missing flow elements', () => {
+    beforeEach(() => {
+      const flow = createInstanceElement(
+        {
+          fullName: 'TestFlow',
+          start: {
+            connector: { targetReference: 'ActionCall' },
+          },
+        },
+        Flow,
+      )
+      flowChange = toChange({ after: flow })
+    })
+    it('should not return any errors', async () => {
+      const errors = await flowReferencedElements([flowChange])
+      expect(errors).toHaveLength(1)
+      const [error] = errors
+      expect(error.severity).toEqual('Error')
+      expect(error.message).toEqual('Reference to missing Flow Element')
+      expect(error.detailedMessage).toEqual(`The Flow Element "${'ActionCall'}" does not exist.`)
+    })
+  })
+  describe('when there are flow elements that are  not referenced', () => {
+    beforeEach(() => {
+      const flow = createInstanceElement(
+        {
+          fullName: 'TestFlow',
+          actionCalls: [
+            {
+              name: 'ActionCall',
+            },
+          ],
+        },
+        Flow,
+      )
+      flowChange = toChange({ after: flow })
+    })
+    it('should not return any errors', async () => {
+      const errors = await flowReferencedElements([flowChange])
+      expect(errors).toHaveLength(1)
+      const [error] = errors
+      expect(error.severity).toEqual('Info')
+      expect(error.message).toEqual('Unused Flow Element')
+      expect(error.detailedMessage).toEqual(`The Flow Element “${'ActionCall'}” isn’t being used in the Flow.`)
+    })
+  })
+  describe('when there are multiple errors in one flow', () => {
+    beforeEach(() => {
+      const flow = createInstanceElement(
+        {
+          fullName: 'TestFlow',
+          start: {
+            connector: { targetReference: 'ActionCall' },
+          },
+          assignments: [
+            {
+              name: 'Assignment',
+              connector: { targetReference: 'Decision' },
+            },
+          ],
+          decisions: [
+            {
+              name: 'Decision',
+              connector: { targetReference: 'RecordCreate' },
+            },
+          ],
+        },
+        Flow,
+      )
+      flowChange = toChange({ after: flow })
+    })
+    it('should create change error per error', async () => {
```

Comment:
```suggestion
    it('should create change error per issue', async () => {
```
Nit:  I think it's confusing to use the term `error` on both sides